realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -381,6 +381,24 @@ public int size() {
         }
     }
 
+    @Override
```

Comment:
@cmelchior and I discussed this. It was determined that we'd fall back to the default `List#contains(...)` implementation when in a non-managed mode. When an object is created from within another query and the developer attempts to find the item via the contains method it will return false (as expected). I'm not sure we need to document that (unless I'm misunderstanding you). I will document the other cases in the javadoc. Thanks for finding this and bringing it up. :boom: